plugins/processors/k8sdecorator/stores/utils.go

```diff
@@ -46,47 +48,46 @@ func parseDeploymentFromReplicaSet(name string) string {
 		return ""
 	}
 	suffix := name[lastDash+1:]
-	if len(suffix) < 3 {
+	if len(suffix) >= 3 && !deploymentUnallowedRegExp.MatchString(suffix){
 		// Invalid suffix if it is less than 3
-		return ""
+		return name[:lastDash]
 	}
 
-	if !stringInRuneset(suffix, kubeAllowedStringAlphaNums) {
-		// Invalid suffix
-		return ""
-	}
-
-	return name[:lastDash]
+	return ""
 }
 
-// get the cronJob name by stripping the last dash following some rules
-// return empty if it is not following the rule
+// Get the cronJob name by stripping the last dash following by the naming convention: JobName-UnixTime
+// based on https://github.com/kubernetes/kubernetes/blob/c4d752765b3bbac2237bf87cf0b1c2e307844666/pkg/controller/cronjob/cronjob_controllerv2.go#L594-L596.
+// Before v1.21 CronJob in Kubernetes has used Unix Time in second; after v1.21 is a Unix Time in Minutes.
+
 func parseCronJobFromJob(name string) string {
 	lastDash := strings.LastIndexAny(name, "-")
+
+	//Return empty since the naming convention is: JobName-UnixTime, if it does not have the "-", meanings the job name is empty
 	if lastDash == -1 {
-		// No dash
 		return ""
 	}
+
 	suffix := name[lastDash+1:]
-	if len(suffix) != 10 {
-		// Invalid suffix if it is not 10 rune
+	suffixInt, err := strconv.ParseInt(suffix, 10, 64)
+
+	if err != nil {
 		return ""
 	}
 
-	if !stringInRuneset(suffix, cronJobAllowedString) {
-		// Invalid suffix
-		return ""
+	//Convert Unix Time In Minutes to Unix Time
+	suffixStringMultiply := strconv.FormatInt(suffixInt*60, 10)
+	//Checking if the suffix is a unix time by checking: the length and contains character
+	//Checking for the length: CronJobControllerV2 is Unix Time in Minutes (7-9 characters) while CronJob is Unix Time (10 characters).
+	//However, multiply by 60 to convert the Unix Time In Minutes back to Unix Time in order to have the same condition as Unix Time
+	if len(suffix) == 10 && !cronJobUnallowedRegexp.MatchString(suffix)  { //Condition for CronJob before k8s v1.21
```

Comment:
I think that having happy paths wrapped in these if-statements and having the default invalid case return at the end is unintuitive. And this gets more confusing when we use the regular expression and check if the string does NOT match the regex. I understand why you would combine the two conditions, but what was the reason for flipping them? 

I think that 
```go
if len(suffix) != 10 || !cronJobUnallowedRegexp.MatchString(suffix) {
  return ""
}
```
is just as readable, and very easy to determine what conditions cause a failure case.

Reply:
Using && would fail faster than using or though.  Since if the first condition is failed, then and would fail the results. However, if for the readable reason, it would be fine by me though.

Reply:
Honestly, I do not think this would work though. These conditions would override the ```suffixStringMultiply``` conditions though.